opentelemetry-api/src/opentelemetry/baggage/propagation/__init__.py

```diff
@@ -49,32 +55,59 @@ def extract(
         )
 
         if not header or len(header) > self._MAX_HEADER_LENGTH:
+            _logger.warning(
+                "Baggage header `%s` exceeded the maximum number of bytes per baggage-string.",
+                header,
+            )
+            return context
+
+        baggage_entries = split(_DELIMITER_PATTERN, header)
+
+        if len(baggage_entries) > self._MAX_PAIRS:
+            _logger.warning(
+                "Baggage header `%s` exceeded the maximum number of list-members",
+                header,
+            )
             return context
 
-        baggage_entries = header.split(",")
-        total_baggage_entries = self._MAX_PAIRS
+        entries = []
         for entry in baggage_entries:
             if len(entry) > self._MAX_PAIR_LENGTH:
+                _logger.warning(
+                    "Baggage entry `%s` exceeded the maximum number of bytes per list-member",
+                    entry,
+                )
+                return context
```

Comment:
I have a question here, If one (or multiple) baggage entries are invalid, do we return the context or skip them and proceed with remaining?

Reply:
This appears to be the implementation others have followed as well.